packages/workbox-runtime-caching/plugins/CacheOkAndOpaquePlugin.mjs

```diff
@@ -0,0 +1,41 @@
+/*
+ Copyright 2016 Google Inc. All Rights Reserved.
+ Licensed under the Apache License, Version 2.0 (the "License");
+ you may not use this file except in compliance with the License.
+ You may obtain a copy of the License at
+
+     http://www.apache.org/licenses/LICENSE-2.0
+
+ Unless required by applicable law or agreed to in writing, software
+ distributed under the License is distributed on an "AS IS" BASIS,
+ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ See the License for the specific language governing permissions and
+ limitations under the License.
+*/
+
+/**
+ * Returns a plugin to allow caching response.ok and opaque responses.
+ *
+ * @return {Object}
+ *
+ * @private
+ */
+class CacheOkAndOpaquePlugin {
+  /**
+   * Return return a response (i.e. allow caching) if the
+   * response is ok (i.e. 200) or is opaque.
+   *
+   * @param {Object} input
+   * @param {Request} input.request
+   * @param {Response} input.response
+   * @return {Response|null}
+   */
+  cacheWillUpdate({request, response}) {
```

Comment:
Do you want to make this static? Or just export an object with `cacheWillUpdate` property by default?